sources/api/netdog/src/cli/write_resolv_conf.rs

```diff
@@ -1,28 +1,187 @@
-        .context(error::GetDnsSettingsSnafu)?;
-    Ok(dns_settings)
+impl InterfaceDNSDropIn {
+    /// Given API DNS settings create an appropriate drop-in for a network interface.
+    fn new(settings: &DnsSettings, is_primary: bool) -> Self {
+        // Default to not using DNS values from DHCP, and do not use the interface's DNS route to
+        // resolve domains not matching other config.  If the interface is the primary interface,
+        // use DNS API settings to direct the appropriate interface configuration.  This matches
+        // existing wicked behavior by ensuring DNS configuration is sourced from settings and the
+        // primary interface only.
+        let mut should_use_dns_from_dhcp = Some(false);
+        let mut should_use_domains_from_dhcp = Some(false);
+        let mut should_be_dns_default_route = Some(false);
+
+        if is_primary {
+            should_use_dns_from_dhcp = Some(!settings.has_name_servers());
+            should_use_domains_from_dhcp = Some(!settings.has_search_domains());
+            should_be_dns_default_route = Some(!settings.has_name_servers());
+        }
+
+        Self {
+            network: Some(NetworkSection {
+                dns_default_route: should_be_dns_default_route,
+            }),
+            dhcp4: Some(Dhcp4Section {
+                use_dns: should_use_dns_from_dhcp,
+                use_domains: should_use_domains_from_dhcp,
+            }),
+            dhcp6: Some(Dhcp6Section {
+                use_dns: should_use_dns_from_dhcp,
+                use_domains: should_use_domains_from_dhcp,
+            }),
+            ipv6_accept_ra: Some(IPv6AcceptRASection {
+                use_dns: should_use_dns_from_dhcp,
+                use_domains: should_use_domains_from_dhcp,
+            }),
+        }
+    }
 }
 
+// If we have DNS name servers or search domain settings from the API, we want to ignore the
+// corresponding values in the DHCP lease.  If we don't, then we want to use the values from DHCP.
+// Toggle this functionality via a networkd interface drop-in.  Also write the global settings from
+// the API as a systemd-resolved drop-in.
+#[cfg(net_backend = "systemd-networkd")]
+fn handle_dns_settings(primary_interface: String) -> Result<()> {
+    let dns_settings = DnsSettings::from_config().context(error::GetDnsSettingsSnafu)?;
+
+    // For each configured interface, create the drop-in directory and file
+    let (maybe_net_config, _) = fetch_net_config()?;
+    if let Some(net_config) = maybe_net_config {
+        for interface in net_config.interfaces() {
+            let interface_drop_in =
+                InterfaceDNSDropIn::new(&dns_settings, interface.to_string() == primary_interface);
+
+            // Remove the colons since the ID might be a MAC address
+            let name = interface.to_string().replace(':', "");
+            let dropin_dir_name = format!("10-{}.network.d", name);
```

Comment:
We define our own convention here - removing the `:` from MAC-address network units? Just checking that we don't need to `systemd-escape` the name instead.

Reply:
We shouldn't need to keep the colons in the names - as long as the unit file and drop-in dir have the same name, networkd doesn't really care what they are called.